lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java

```diff
@@ -28,15 +30,29 @@ public void shouldThrowWhenRequestingSignWithoutAlgorithm() throws Exception {
 
     @SuppressWarnings("Convert2Diamond")
     @Test
-    public void shouldAddHeader() throws Exception {
+    public void shouldAddHeaderClaim() throws Exception {
         Map<String, Object> header = new HashMap<String, Object>();
         header.put("asd", 123);
         String signed = JWTCreator.init()
                 .withHeader(header)
                 .sign(Algorithm.HMAC256("secret"));
 
         assertThat(signed, is(notNullValue()));
-        assertThat(TokenUtils.splitToken(signed)[0], is("eyJhbGciOiJIUzI1NiIsImFzZCI6MTIzfQ"));
+        String[] parts = signed.split("\\.");
+        String headerJson = new String(Base64.decodeBase64(parts[0]), StandardCharsets.UTF_8);
+        assertThat(headerJson, JsonMatcher.hasEntry("asd", 123));
+    }
+
+    @Test
+    public void shouldAddKeyId() throws Exception {
+        String signed = JWTCreator.init()
+                .withKeyId("56a8bd44da435300010000015f5ed")
+                .sign(Algorithm.HMAC256("secret"));
+
+        assertThat(signed, is(notNullValue()));
+        String[] parts = signed.split("\\.");
+        String headerJson = new String(Base64.decodeBase64(parts[0]), StandardCharsets.UTF_8);
+        assertThat(headerJson, JsonMatcher.hasEntry("kid", "56a8bd44da435300010000015f5ed"));
```

Comment:
since this code is repeated a few times, maybe we could add a matcher, something like `JWTMatcher.hasHeaderEntry` ? Or at least a method to wrap the split, base64decode of parts[0] and the hasEntry.